src/openvpn/ssl_openssl.c

```diff
@@ -959,9 +1162,14 @@ tls_ctx_load_cert_file(struct tls_root_ctx *ctx, const char *cert_file,
     {
         in = BIO_new_mem_buf((char *) cert_file, -1);
     }
+    else if (cert_uri_supported())
+    {
+        tls_ctx_load_cert_uri(ctx, cert_file);
+        return;
+    }
     else
     {
-        in = BIO_new_file(cert_file, "r");
+        in = BIO_new_file((char *) cert_file, "r");
     }
 
     if (in == NULL)
```

Comment:
Maybe move this following part into a separate function like `tls_ctx_load_cert_bio` and then used that afer the `in =` in these two branches that use that? That will make logic here easier to understand and indicate that it is similar to the `tls_ctx_load_cert_uri` function in what it is doing?

Reply:
I like the idea but we may lose the ability to print the file name in the error message if loading fails. May be I can name it as "tls_ctx_load_cert_pem()" and pass the filename itself instead of bio. Let me see..

Reply:
Good idea. Instead of passing "in" branched to tls_ctx_load_cert_pem_file() when URI is not supported or if inline. That allows us to continue printing file name and inline flag in the error messages.